src/components/PostModal.jsx

```diff
@@ -0,0 +1,61 @@
+import React,{useState,useRef} from "react";
+import { useSelector,useDispatch } from "react-redux";
+import { useOutsideClick } from "../hooks/useOutsideClick";
+import { postModalOff } from "../redux/features/postSlice";
+import { addPost, editPost } from "../utils/post-utils/post-services";
+const calculateContent=(selectedPost)=>{
+  if(Object.keys(selectedPost).length===0 && selectedPost.constructor===Object){
+    return ''
+  }else return selectedPost.content
+}
+
+const PostModal=()=>{
+    
+    const user = useSelector((state) => state.auth.loggedInUser);
+    const selectedPost=useSelector(state=>state.posts.selectedPost)
+    const refr=useRef(null)
+    const [content,setContent]=useState(calculateContent(selectedPost))
+    
+    const dispatch=useDispatch()
+    const clickHandler=()=>{
+        if(content){
+          if(!selectedPost){
+          const post={content,comments:[],userProfile:user.userProfile}
+            setContent('')
+            dispatch(addPost(post))
+            dispatch(postModalOff())
+          }else {
+          const post={content,comments:[],userProfile:user.userProfile,_id:selectedPost._id}
+            setContent('')
+            dispatch(editPost(post))
+            dispatch(postModalOff())
+          }
+        
+        }
+    }
+    useOutsideClick(refr,()=>dispatch(postModalOff()))
+    return (
+      <div className="fixed min-h-screen min-w-[100vw] top-0 left-0 grid place-items-center bg-[rgba(0,0,0,0.5)] z-20" >
+        <div ref={refr}className="border-solid border-2 border-black bg-lightOrange p-6 rounded shadow-neu w-fit min-w-[30rem]  h-fit">
+        <div className="flex gap-2">
+          <img
+            className="w-12 rounded-full object-cover h-12 "
+            src={user.userProfile}
+            alt=""
```

Comment:
You can give the alt text as profile pic here, as this image contains the profile pic of the user right?

Reply:
Yes